### ISSUE TITLE
Dotty 0.20.0-RC1: Cannot write a given alias for a function

### ISSUE THREAD

**Issue by J-mie6** [lampepfl/dotty#7515]
## minimized code
```Scala
given (Int => Int) = identity[Int]
```
Produces three errors:
end of statement expected but '(' found
';' expected, but identifier found
anonymous instance must implement a type or have at least one extension method

## expectation
The code above should work, just like the below does:

```scala
type IntFunc = Int => Int
given IntFunc = identity[Int]
```

**Comment by nicolasstucki** [comment]
I am not sure if the current syntax is supposed to allow it.

**Comment by J-mie6** [comment]
It worked as of 0.19.0-RC1, which was the release that I would have expected that to have been banned in... Seems a shame to either use type aliases or I assume `Function[Int, Int]` would work too.

**Comment by odersky** [comment]
This will be fixed by #7557.
